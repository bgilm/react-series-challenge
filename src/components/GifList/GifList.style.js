import styled from 'styled-components'

export const GifWrapper = styled.div`
  width: 100%;
  height: 100%;
`

export const GifStyledList = styled.ul`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  height: 2100px;
  width: 1000px;
  padding: 5px;
  margin: auto;
`

export const GifItemWrapper = styled.div`
  margin: 5px;
  width: fit-content;
  height: fit-content;
`
export const StyledButton = styled.div`
  position: absolute;
  color: ${props => props.isFavorite ? 'red' : 'white'};
  cursor: pointer;

  :hover {
    color:red;
  }
`
