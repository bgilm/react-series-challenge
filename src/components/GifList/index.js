import React, { Component } from 'react'
import {
  GifWrapper,
  GifStyledList
} from './GifList.style'
import GifItem from '../GifItem'
import Menu from '../Menu'
import SearchBar from '../SearchBar'

class GifList extends Component {

  componentWillMount() {
    if (this.props.requestGifs){
      this.props.requestGifs()
    }
  }

  checkIfFavorite(id) {
    const fav = this.props.favoriteGifs.find(gif => gif.id === id)
    return (fav) ? true : false
  }

  renderGifsIfExist() {
    if(Array.isArray(this.props.gifs)){
      return(
        <GifStyledList>
          {this.props.gifs.map(gif => {
            const isFavorite = this.checkIfFavorite(gif.id)
            return(
              <GifItem
                key={gif.id}
                gif={gif}
                setFavorite={this.props.setFavorite}
                isFavorite={isFavorite}
              />
            )
          })}
        </GifStyledList>
      )
    }
  }
  render() {
    return(
      <GifWrapper>
        <SearchBar />
        <Menu location={this.props.location.pathname} />
        {this.renderGifsIfExist()}
      </GifWrapper>
    )
  }
}

export default GifList
