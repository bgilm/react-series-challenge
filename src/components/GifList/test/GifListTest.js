/**
 * @jest-environment node
 */

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import GifList from '..';
const props = {
  location: {pathname: 'test'},
  match: {params: 'test'},
  gifs: [],
  searchGifs: jest.fn(),
  searchFavs: jest.fn()
}

describe('GifList', () => {
  it('should render correctly', () => {
    const output = shallow(
      <GifList isFavorite={true} gif={{title:'title'}} setFavorite={() => {}} {...props} />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should be defined', () => {
   expect(GifList).toBeDefined();
  });
});
