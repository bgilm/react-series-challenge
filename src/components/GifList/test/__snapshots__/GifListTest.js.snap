// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifList should render correctly 1`] = `
<styled.div>
  <withRouter(SearchBar)
    searchGifs={[MockFunction]}
  />
  <Menu
    location="test"
  />
  <styled.ul />
</styled.div>
`;
